### PR TITLE
fix: add geocoding radius functionality

### PR DESCRIPTION
**Pull request by ludtkemorgan**
# Pull Request Template

## Issue Overview

This PR addresses #issue

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR includes two recent changes from the old backend:
- Geocoding radius functionality - https://github.com/bloom-housing/bloom/pull/3718
- Email send change - https://github.com/bloom-housing/bloom/pull/3731

## How Can This Be Tested/Reviewed?

The geocoding functionality can be fully tested end to end with the following steps:
1. Create a new preference in the partner site that has geocoding enabled and radius selected and add it to an open listing
2. Go to the public site and apply to the listing - make sure you fill out the address for that preference.
3. Check the DB for that application and the preference extra data should have accurate information (true/false/Unknown)

The email send cannot be tested since the listing changes are still in a different PR.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [n/a] I have reviewed the changes in a desktop view
- [n/a] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [n/a] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend_new/src/enums/shared/geocoding-values.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum GeocodingValues {
```
Comment (emilyjablonski) [inline backend_new/src/enums/shared/geocoding-values.ts:1]: Curious here why we don't use true/false/unknown types?

File: backend_new/src/services/application.service.ts [hunk 2]
```diff
@@ -373,6 +375,19 @@ export class ApplicationService {
...
 
+    // Calculate geocoding preferences after save and email sent
+    if (listing.jurisdictions?.enableGeocodingPreferences) {
+      try {
+        void this.geocodingService.validateGeocodingPreferences(
+          mapTo(Application, rawApplication),
+          mapTo(Listing, listing),
+        );
+      } catch (e) {
+        // If the geocoding fails it should not prevent
```
Comment (emilyjablonski) [inline backend_new/src/services/application.service.ts:386]: Just a nit but not 100p clear what should not be prevented here

File: backend_new/src/enums/shared/geocoding-values.ts (hunk 1 above)
Comment (ludtkemorgan) [reply backend_new/src/enums/shared/geocoding-values.ts:1]: Can you explain how that would look?

Comment (emilyjablonski) [reply backend_new/src/enums/shared/geocoding-values.ts:1]: Potentially that `verifyRadius` would just return `boolean | unknown`

Comment (ludtkemorgan) [reply backend_new/src/enums/shared/geocoding-values.ts:1]: Unfortunately you can't return `unknown` from a function. I was hoping that this GeocodingValues could also be read for the CSV function as well. But I'll keep digging into this

File: backend_new/src/services/application.service.ts (hunk 2 above)
Comment (ludtkemorgan) [reply backend_new/src/services/application.service.ts:386]: Looks like I started writing a sentence and then failed to finish it. I'll update this

**Comment by ludtkemorgan** [comment]
@emilyjablonski have you had a chance to re-review this?

**Review by emilyjablonski — APPROVED** [review]

**Pull request merged**